src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,161 @@
+
+export const createGetAppMenu =
+  (services: ProfileProviderServices): RootProfileProvider['profile']['getAppMenu'] =>
+  (prev) =>
+  (params) => {
+    const prevValue = prev(params);
+
+    return {
+      appMenuRegistry: (registry) => {
+        // Register custom link actions
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.datasets', {
+          defaultMessage: 'Data sets',
+        }),
+        testId: 'discoverAppMenuDatasetQualityLink',
+        onClick: ({ onFinishAction }) => {
+          const refresh = timefilter.getRefreshInterval();
+          const { from, to } = timefilter.getTime();
+
+          dataQualityLocator.navigate({
+            filters: {
+              timeRange: {
+                from: from ?? 'now-24h',
+                to: to ?? 'now',
+                refresh,
+              },
+            },
+          });
+
+          onFinishAction();
+        },
+      },
+    });
+  }
+};
+
+const registerCustomThresholdRuleAction = (
+  registry: AppMenuRegistry,
+  { data, triggersActionsUi }: ProfileProviderServices,
+  { dataView }: AppMenuExtensionParams
+) => {
+  registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+    id: AppMenuActionId.createRule,
+    type: AppMenuActionType.custom,
+    order: 101,
+    controlProps: {
+      label: i18n.translate('discover.observabilitySolution.appMenu.customThresholdRule', {
+        defaultMessage: 'Create custom threshold rule',
+      }),
+      iconType: 'visGauge',
+      testId: 'discoverAppMenuCustomThresholdRule',
+      onClick: ({ onFinishAction }) => {
+        const index = dataView?.toMinimalSpec();
+        const { filters, query } = data.query.getState();
+
+        return triggersActionsUi.getAddRuleFlyout({
+          consumer: 'logs',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          canChangeTrigger: false,
+          initialValues: {
+            params: {
+              searchConfiguration: {
+                index,
+                query,
+                filter: filters,
+              },
+            },
+          },
+          onClose: onFinishAction,
+        });
+      },
+    },
+  });
+};
+
+const registerCreateSLOAction = (
```

Comment:
I ran into an issue where the request failed when I tried to create an SLO, seemingly because of the filter I was using in Discover:
![CleanShot 2024-11-07 at 23 58 17@2x](https://github.com/user-attachments/assets/6b69b105-8531-40e7-a3d8-b14466950d44)

Reply:
I reproduced it but it looks like a bug on the SLOs end, we are just embedding the creation flyout. 
Could you open an issue for the SLOs team please?

Reply:
Hey Davis, I tried this on Logs Explorer too and it is affected in the same way.
As you thought, the expectation is that the filters don't have the `$state` property, which is reason why it fails validating the payload sent to the server and fails the SLO creation.

I stripped the property client side and it works fine now

Reply:
Nice, I confirmed it works now!